js/test/utils.test.js

```diff
@@ -1,67 +1,71 @@
+        const res = utils.lastBytes(lastBytes[i].input.bytes, lastBytes[i].input.num);
+        const arraysAreEqual = utils.typedArraysAreEqual(res, lastBytes[i].output);
+        assert.isTrue(arraysAreEqual);
+      }
     });
 
     it('errors if slice is larger than the bytearray', () => {
-      try {
-        utils.lastBytes(utils.deserializeHex('0x00'), 2);
-        assert(false, 'expected an errror');
-      } catch (e) {
-        assert.include(e.message, 'Slice must not use negative indexes');
+      for (let i = 0; i < lastBytesError.length; i += 1) {
+        try {
+          utils.lastBytes(lastBytesError[i].input.bytes, lastBytesError[i].input.num);
+          assert(false, 'expected an errror');
+        } catch (e) {
+          assert.include(e.message, lastBytesError[i].errorMessage);
+        }
       }
     });
   });
 
   describe('reverseEndianness', () => {
     it('reverses endianness', () => {
-      let res;
-      let arraysAreEqual;
-      res = utils.reverseEndianness(utils.deserializeHex('0x00112233'));
-      arraysAreEqual = utils.typedArraysAreEqual(res, utils.deserializeHex('0x33221100'));
-      assert.isTrue(arraysAreEqual);
-
-      res = utils.reverseEndianness(utils.deserializeHex('0x0123456789abcdef'));
-      arraysAreEqual = utils.typedArraysAreEqual(res, utils.deserializeHex('0xefcdab8967452301'));
-      assert.isTrue(arraysAreEqual);
+      for (let i = 0; i < reverseEndianness.length; i += 1) {
+        const res = utils.reverseEndianness(reverseEndianness[i].input);
+        const arraysAreEqual = utils.typedArraysAreEqual(res, reverseEndianness[i].output);
+        assert.isTrue(arraysAreEqual);
+      }
     });
   });
 
   describe('#bytesToUint', () => {
     it('converts big-endian bytes to integers', () => {
-      let res = utils.bytesToUint(utils.deserializeHex('0x00'));
-      assert.equal(res, BigInt(0));
-
-      res = utils.bytesToUint(utils.deserializeHex('0xff'));
-      assert.equal(res, BigInt(255));
-
-      res = utils.bytesToUint(utils.deserializeHex('0x00ff'));
-      assert.equal(res, BigInt(255));
-
-      res = utils.bytesToUint(utils.deserializeHex('0xff00'));
-      assert.equal(res, BigInt(65280));
-
-      res = utils.bytesToUint(utils.deserializeHex('0x01'));
-      assert.equal(res, BigInt(1));
-
-      res = utils.bytesToUint(utils.deserializeHex('0x0001'));
-      assert.equal(res, BigInt(1));
-
-      res = utils.bytesToUint(utils.deserializeHex('0x0100'));
-      assert.equal(res, BigInt(256));
-
+      let res;
+      for (let i = 0; i < 7; i += 1) {
+        res = utils.bytesToUint(bytesToUint[i].input);
+        assert.equal(res, BigInt(bytesToUint[i].output));
+      }
       // max uint256: (2^256)-1
-      res = utils.bytesToUint(utils.deserializeHex('0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'));
+      res = utils.bytesToUint(bytesToUint[7].input);
+      // cannot store this value in store and have it test correctly
```

Comment:
point taken. instead of putting the input in and not the output, put neither in, and have both input and output special cased here